website/blog/2022-07-25-coreapp.md

```diff
@@ -0,0 +1,128 @@
+
+The `launched` callback above gives you an output `RNCoreApp` structure, where you can set a number of parameters, as well as the `data` parameter that you passed in when you called the `CoreApp` API.
+
+The contents of the `RNCoreApp` structure can be fully set via the JSON configuration file, so next we'll take a look at what is in this JSON configuration file.
+
+## `CoreApp` schema
+
+Below are the properties you can set in your app.config.json, with their default values.
+These properties correspond to properties on the ReactInstanceSettings type.
+```json
+{
+  "jsBundleFile": "index.windows",
+  "bundleRootPath": "ms-appx:///Bundle/",
+  "componentName": null, // Required, this is your App's component name
+  "useWebDebugger": true,
+  "useFastRefresh": true,
+  "useDeveloperSupport": true,
+  "useDirectDebugger": false,
+  "requestInlineSourceMap": true,
+  "enableDefaultCrashHandler": false,
+  "debuggerPort": 9229,
+  "sourceBundlePort": 8081,
+  "sourceBundleHost": "localhost",
+  "jsEngine": "chakra", // possible values: "chakra", "hermes"
+  "viewName": null, // adds an optional title to the window
+  "nativeModules": [
+    {
+      // this corresponds to the DLL that hosts the native module,
+      // or the name of the app's exe (or null) if the module is locally defined
+      "moduleContainer": null,  
+
+      // the name of the factory function to call to produce a ReactPackageProvider for
+      // the module; see Using native modules below.
+      "factory": null,  
+      
+    }
+  ],
+  "properties": { 
+    // these are all optional. They correspond to properties that will get set in the
+    // ReactPropertyBag for the instance. Some examples:
+    "someString": "string value",
+    "someNumber": 42.5,
+    "someBoolean": true,
+    "namespace1.namespace2.foo": 22
+  }
+}
+```
+
+For building in Debug mode, your app will usually only need to set a few properties: `componentName`, and the native modules it uses.
+
+## Using native modules
+There are a couple of ways that a `CoreApp` can load native modules.
+
+The simplest way is using the `RNCoreAppStartFromConfigJson` API. As we saw above, this API allows us to pass an optional DLL name to load and a plain C function to call in that DLL, to produce the `ReactPackageProvider` for the module.
+Here's what this function would look like:
+```cpp
+extern "C" __declspec(dllexport) void *__cdecl MySpecialPackageProvider() {
+  auto provider = winrt::make<MyModulePackageProvider>();
+  void *abi{nullptr};
+  winrt::copy_to_abi(provider, abi);
+  return abi;
+}
+```
+
+This snippet will declare a plain C function (this is important so that C++ name mangling mechanism doesn't come into play, which is required to be able to use the function name in the JSON file), and export it from the DLL (or the application's EXE). It takes no parameters, and simply return a pointer to the package provider for the module.
```

Comment:
The formatting from here out looks funky - might just be strangely rendered in GitHub, but probably want to double check that it looks good when published.

Reply:
validated this looks ok on the website, lmk if you disagree:
<img width="749" alt="image" src="https://user-images.githubusercontent.com/22989529/180899113-bbc7c5c8-6da1-47c5-9882-6b2a7e67c712.png">